### ISSUE TITLE
Global uniform does not exist error on project startup

### ISSUE THREAD

**Issue by EspeuteClement** [godotengine/godot#92088]
### Tested versions

v4.3.dev6.official [89850d553]

### System information

Godot v4.3.dev6 - Windows 10.0.19045 - Vulkan (Mobile) - dedicated NVIDIA GeForce GTX 1660 Ti (NVIDIA; 31.0.15.3699) - AMD Ryzen 5 2600 Six-Core Processor (12 Threads)

### Issue description

When opening a project that uses a global uniform of type sampler 2d, an error is shown in the godot console output `Global uniform 'noise' does not exist. Create it in Project Settings.` while the uniform actually exist in the project.

This was also reported in a new comment on a closed issue here : https://github.com/godotengine/godot/issues/57200#issuecomment-2097458585

### Steps to reproduce

1. Create a project using the Mobile render target, create a new global uniform named noise of type sampler 2d, set it to a noise texture
2. Create the following shader
```
shader_type canvas_item;

global uniform sampler2D noise;

void fragment() {
	COLOR = texture(noise, UV);
}
```
3. Assing the shader to a sprite2d node, save the scene, reload the editor
4. Observe the error messages in the ouput console

### Minimal reproduction project (MRP)

[globaluniform.zip](https://github.com/godotengine/godot/files/15366125/globaluniform.zip)

**Comment by matheusmdx** [comment]
I did the test and i can confirm the error happens, also this a regression caused by #60965 in v4.3.dev1

![image](https://github.com/godotengine/godot/assets/44306054/03a80f7d-d80b-4ce6-b05a-37c320e65ca1)

A thing i noticied while i did my test is another bug: When i create a resource directly in the project settings option, the resource is never saved inside the project file, using notepad++ i see the file changed but when i reload nothing changes and the resource is not stored. Now if i use a resource already saved in a separated file, the project file updates correctly. Revert #60965 doesn't solve this part.

https://github.com/godotengine/godot/assets/44306054/3acb76f8-71e3-4800-93bf-6969cbe01d4b

**Comment by akien-mga** [comment]
CC @DarkMessiah @KoBeWi

**Comment by matheusmdx** [comment]
I tested compiling from master daa81bbb7 and this error was already solved by #91414 

![image](https://github.com/godotengine/godot/assets/44306054/99c4edcd-8448-48e5-83b6-6be46cfaa406)

The resources not being saved in the global shaders problem persists, should a separated issue be opened for that?

**Comment by akien-mga** [comment]
> I tested compiling from master https://github.com/godotengine/godot/commit/daa81bbb7d1c6d75d1711595604178ee62a5801d and this error was already solved by https://github.com/godotengine/godot/pull/91414

Thanks, closing as fixed by #91414.

> The resources not being saved in the global shaders problem persists, should a separated issue be opened for that?

Yes, that sounds relevant. Could you open one if you have a reproduction project?
